drivers/soundwire/bus_type.c

```diff
@@ -108,7 +109,10 @@ static int sdw_drv_probe(struct device *dev)
 
 	ret = drv->probe(slave, id);
 	if (ret) {
-		dev_err(dev, "Probe of %s failed: %d\n", drv->name, ret);
+		name = drv->name;
+		if (!name)
+			name = drv->driver.name;
+		dev_err(dev, "Probe of %s failed: %d\n", name, ret);
```

Comment:
Maybe add a macro to look up "sdw_driver_name(sdw_driver *driver)"? I'd guess this is needed in many places. Or maybe just drop the top-level name.

Reply:
i'll see what Vinod wants when upstreaming, this is a cosmetic detail.